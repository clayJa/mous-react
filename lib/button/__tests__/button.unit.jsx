import * as renderer from 'react-test-renderer';
import * as React from 'react';
import Button from '../button';
import {mount} from 'enzyme';

describe('button', () => {
    it('render successfully', () => {
        const json = renderer.create(<Button />).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick',() => {
        const fn = jest.fn()
        const component = mount(<Button onClick={fn}/>)
        component.find('button').simulate('click');
        expect(fn).toBeCalled()
    })
    it('props mode',() => {
        const component = mount(<Button mode={`danger`}/>)
        expect(component.find('button').hasClass('rainbow-button-danger')).toEqual(true)
    })
    it('props loading',() => {
        const component = mount(<Button loading={`danger`}/>)
        expect(component.find('button').children().find('svg').hasClass('loading')).toEqual(true)
    })
});
