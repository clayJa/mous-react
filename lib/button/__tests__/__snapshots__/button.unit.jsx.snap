// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button render successfully 1`] = `
<button
  className="mous-button-normal mous-button"
>
  
</button>
`;
